src/core/secondaryAdapters/officialOnyxiaApiClient.ts

```diff
@@ -1,10 +1,14 @@
 import type {
-    OnyxiaApiClient,
+    Catalog,
```

Comment:
This type shouldn't be declared outside of this file nor exported.  

The type `Catalog` is the 'de facto' type that is returned by the Java version of the Onyxia API.  We never use it outside of this adapter.  

The Java implementation of the Onyxia API is not actively maintained, there is a lot of legacy information and inconsistent naming in the returned values.  

In `ports/OnyxiaApiClient.ts` we declare the types Onyxia-web actually uses.  
In an idal world the type of the API would match the type used internally by Onyxia-web but this would require having a Node.js implementation of the API.  

Beside, exporting this type break the Clean Architecture philosophy.  
Imagine we have a new implementation of Onyxia-API like [gonixia-api](https://github.com/mgrzybek/gonyxia-api) that emerges.  
It won't probably return exactly the same object type as the Java implementation but probably enough so that we can write an adapter that fullfill the contract specified in `ports/OnyxiaApiClient.ts`.

Reply:
I hard-coded the type returned by the API in order to allow Onyxia-web to redefine as it wishes the type expected in the promise.

Was that the point of the review ?

Reply:
My bad for this... I missread the diff.